### PR TITLE
[BugFix] BE crash when ingesting nested array from orc

### LINKED ISSUE #16365: BE crash when ingesting nested array from orc
### Steps to reproduce the behavior (Required)
1. CREATE TABLE `t1` (
  `c0` int(11) NULL COMMENT "",
  `c1` ARRAY<ARRAY<varchar(1000000)>> NOT NULL COMMENT ""
) ENGINE=OLAP 
DUPLICATE KEY(`c0`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`c0`) BUCKETS 3 
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false",
"compression" = "LZ4"
);
2. create orc file with issue_16365_generate.py, upload it to hdfs
in this orc file, c1's array nested depth limit is 1, not 2
4. LOAD LABEL duplicate_tab (DATA INFILE( "xxx.orc") INTO TABLE `t1` FORMAT AS `orc` (col_int8, col_json_list_list) SET (c0=col_int8,c1=col_json_list_list))  WITH BROKER hdfs_broker ("username"="xx","password"="xx") PROPERTIES("timeout"="300");

### Expected behavior (Required)
load fail

### Real behavior (Required)
BE crash

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
e6922b22a915fc7f4fabd3101e813e21c4735dfc

### PR DESCRIPTION
**Pull request by xiaoyong-z**
Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16365

## Problem Summary(Required) ：

If we create a table with a nested `array<array<int>>`, and ingest data into this table with a nested `array<int>`, then BE will crash.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: be/test/exec/test_data/orc_scanner/issue_16365_generate.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright 2021-present StarRocks, Inc. All rights reserved.
```
Comment (chaoyli) [inline be/test/exec/test_data/orc_scanner/issue_16365_generate.py:1]: You can place the orc file in the directory.

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by xiaoyong-z** [comment]
run starrocks_be_unittest

**Review by trueeyu — APPROVED** [review]

**Comment by trueeyu** [comment]
Why add the orc file?

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**